⭐ add artifacts for JOSS integration (#407)
Cloud-Drift/clouddrift: paper.md

---
title: 'Clouddrift: a Python package to accelerate the use of Lagrangian data for atmospheric, oceanic, and climate sciences'
tags:
  - Python
  - climate
  - ocean
  - atmosphere
  - ragged array
authors:
  - name: Shane Elipot
    orcid: 0000-0001-6051-5426
    equal-contrib: true
    affiliation: 1
  - name: Philippe Miron
    orcid: 0000-0002-8520-6221
    equal-contrib: true
    affiliation: 2
  - name: Milan Curcic
    orcid: 0000-0002-8822-7749
    equal-contrib: true
    affiliation: 1
  - name: Kevin Santana
    orcid: 0009-0003-8383-1212
    equal-contrib: true
    affiliation: 1
  - name: Rick Lumpkin
    orcid: 0000-0002-6690-1704
    equal-contrib: true
    affiliation: 3

affiliations:
 - name: Rosenstiel School of Marine, Atmospheric, and Earth science, University of Miami
   index: 1
 - name: Florida State University
   index: 2
 - name: NOAA Atlantic Oceanographic and Meteorological Laboratory
   index: 3
date: 10 April 2024
bibliography: paper.bib

---

# Summary

> Lagrangian data in Earth sciences are unique because they do not conform to established standards related to dimensions, coordinates, and organizational structures. In addition, because they convolve spatial and temporal information, Lagrangian data necessitate specific processing and analysis tools for their scientific and operational exploitations. The Clouddrift Python library offers tools to process and analyze Lagrangian data with an emphasis on the ragged array representation.

Reviewer: ```suggestion
Lagrangian data in Earth sciences are unique because they do not conform to established standards related to dimensions, coordinates, and organizational structures. In addition, because they convolve spatial and temporal information, Lagrangian data necessitate specific processing and analysis tools for their scientific and operational exploitations. The Clouddrift Python library addresses those challenges by offering tools to process and analyze Lagrangian data with an emphasis on the ragged array representation.
```

# Statement of need

> In Earth, Ocean, Geo-, and Atmospheric Science, Eulerian typically refers to a type of data  acquired or simulated at a particular fixed point or region in space. Eulerian data often materialize on spatiotemporal grids with monotonic coordinates (e.g. latitude, longitude, depth, time) for which popular Python tools such as [xarray](https://docs.xarray.dev/en/stable/) are naturally suited. Eulerian data are often contrasted with Lagrangian data. Such data refers to information acquired by observing platforms drifting with a flow they are embedded within, but also more broadly refers to the data originating from uncrewed platforms, vehicles, virtual particles, atmospheric phenomena such as tropical cyclones, and even animals that gather data along their natural but complicated paths.  Because such paths traverse both spatial and temporal dimensions, Lagrangian data often convolve spatial and temporal information that cannot always and readily be organized, cataloged, and stored in common data structures and file formats with the help of common libraries and standards. As an example, the concepts of dimensions and coordinates for Lagrangian data are ambiguous and not clearly established. As such, for both data generators and data users, Lagrangian data present challenges that the Clouddrift Python library aims to overcome. 

Reviewer: ```suggestion
In Earth, Ocean, Geo-, and Atmospheric Science, Eulerian typically refers to a type of data acquired or simulated at a particular fixed point or region in space. Eulerian data often materialize on fixed spatiotemporal grids with monotonic coordinates (e.g. latitude, longitude, depth, time) for which popular Python tools such as [xarray](https://docs.xarray.dev/en/stable/) are naturally suited. Eulerian data are often contrasted with Lagrangian data. Such data refers to information acquired by observing platforms drifting with a flow they are embedded within, but also more broadly refers to the data originating from uncrewed platforms, vehicles, virtual particles, atmospheric phenomena such as tropical cyclones, and even animals that gather data along their natural but complicated paths. Because such paths traverse both spatial and temporal dimensions, Lagrangian data often convolve spatial and temporal information that cannot consistently and readily be organized, cataloged, and stored in common data structures and file formats with the help of common libraries and standards. As an example, the concepts of dimensions and coordinates for Lagrangian data are ambiguous and not clearly established. As such, for both data generators and data users, Lagrangian data present challenges that the Clouddrift Python library aims to overcome. 
```

# Scope and key features

The scope of the Clouddrift library includes: 

> 1. **Working with contiguous ragged array representations of data, wether they originate from geosciences or any other field**. Ragged array representations are typically useful when the data lengths of the instances of a feature (variable) are not all equal. With such representations the data for each features are stored contiguously in memory, and the number of elements that each feature has is contained in a count variable which Clouddrift calls *rowsize*. 

Reviewer: ```suggestion
1. **Working with contiguous ragged array representations of data, whether they originate from geosciences or any other field**. Ragged array representations are useful when the data lengths of the instances of a feature (variable) are not all equal. With such representations the data for each feature are stored contiguously in memory, and the number of elements that each feature has is contained in a count variable which Clouddrift calls *rowsize*. 
```

> ![Ragged array representation for Lagrangian data.\label{fig:ragged_array}](./docs/img/ragged_array.png)       

Reviewer: we have to include the image in the path (or include the folder).

Reviewer 2: which path?

Reviewer: you have the figure in `docs/img/ragged_array.png` here. Currently there is only `paper.md` and `paper.bib`.

---
title: 'Clouddrift: a Python package to accelerate the use of Lagrangian data for atmospheric, oceanic, and climate sciences'
tags:
  - Python
  - climate
  - ocean
  - atmosphere
  - ragged array
authors:
  - name: Shane Elipot
    orcid: 0000-0001-6051-5426
    equal-contrib: true
    affiliation: 1
  - name: Philippe Miron
    orcid: 0000-0002-8520-6221
    equal-contrib: true
    affiliation: 2
  - name: Milan Curcic
    orcid: 0000-0002-8822-7749
    equal-contrib: true

>     affiliation: 1

Reviewer 3: ```suggestion
    affiliation: 1, 4
```

(not sure if this is the correct syntax)

  - name: Kevin Santana
    orcid: 0009-0003-8383-1212
    equal-contrib: true
    affiliation: 1
  - name: Rick Lumpkin
    orcid: 0000-0002-6690-1704
    equal-contrib: true
    affiliation: 3

affiliations:
 - name: Rosenstiel School of Marine, Atmospheric, and Earth science, University of Miami
   index: 1
 - name: Florida State University
   index: 2
 - name: NOAA Atlantic Oceanographic and Meteorological Laboratory

>    index: 3

Reviewer 3: ```suggestion
   index: 3
 - name: Frost Institute for Data Science and Computing, University of Miami
   index: 4
```

---
title: 'Clouddrift: a Python package to accelerate the use of Lagrangian data for atmospheric, oceanic, and climate sciences'
tags:
  - Python
  - climate
  - ocean
  - atmosphere
  - ragged array
authors:
  - name: Shane Elipot
    orcid: 0000-0001-6051-5426
    equal-contrib: true
    affiliation: 1
  - name: Philippe Miron
    orcid: 0000-0002-8520-6221
    equal-contrib: true
    affiliation: 2
  - name: Milan Curcic
    orcid: 0000-0002-8822-7749
    equal-contrib: true
    affiliation: 1, 4
  - name: Kevin Santana
    orcid: 0009-0003-8383-1212
    equal-contrib: true
    affiliation: 1
  - name: Rick Lumpkin
    orcid: 0000-0002-6690-1704
    equal-contrib: true
    affiliation: 3

affiliations:
 - name: Rosenstiel School of Marine, Atmospheric, and Earth science, University of Miami
   index: 1
 - name: Florida State University
   index: 2
 - name: NOAA Atlantic Oceanographic and Meteorological Laboratory
   index: 3
 - name: Frost Institute for Data Science and Computing, University of Miami
   index: 4
date: 10 April 2024
bibliography: paper.bib

---

# Summary

> Lagrangian data in Earth sciences are unique because they do not conform to established standards related to dimensions, coordinates, and organizational structures. In addition, because they convolve spatial and temporal information, Lagrangian data necessitate specific processing and analysis tools for their scientific and operational exploitations. The Clouddrift Python library addresses those challenges by offering tools to process and analyze Lagrangian data with an emphasis on the ragged array representation.

Reviewer 3: Simpler language

```suggestion
Lagrangian data in Earth sciences are unique because they do not conform to established standards regarding dimensions, coordinates, and organizational structures. In addition, because they convolve spatial and temporal information, Lagrangian data need specific processing and analysis tools for their scientific and operational use. The Clouddrift Python library addresses these challenges by offering tools to process and analyze Lagrangian data with an emphasis on the ragged array representation.
```

# Statement of need

> In Earth, Ocean, Geo-, and Atmospheric Science, Eulerian typically refers to a type of data acquired or simulated at a particular fixed point or region in space. Eulerian data often materialize on fixed spatiotemporal grids with monotonic coordinates (e.g. latitude, longitude, depth, time) for which popular Python tools such as [xarray](https://docs.xarray.dev/en/stable/) are naturally suited. Eulerian data are often contrasted with Lagrangian data. Such data refers to information acquired by observing platforms drifting with a flow they are embedded within, but also more broadly refers to the data originating from uncrewed platforms, vehicles, virtual particles, atmospheric phenomena such as tropical cyclones, and even animals that gather data along their natural but complicated paths. Because such paths traverse both spatial and temporal dimensions, Lagrangian data often convolve spatial and temporal information that cannot consistently and readily be organized, cataloged, and stored in common data structures and file formats with the help of common libraries and standards. As an example, the concepts of dimensions and coordinates for Lagrangian data are ambiguous and not clearly established. As such, for both data generators and data users, Lagrangian data present challenges that the Clouddrift Python library aims to overcome. 

Reviewer 3: Simpler language

```suggestion
In Earth, Ocean, Geo-, and Atmospheric Science, "Eulerian data" typically refers to data acquired or simulated at a particular fixed point or region in space. Eulerian data are defined on fixed spatiotemporal grids with monotonic coordinates (e.g. latitude, longitude, depth, time) for which popular Python tools such as [Xarray](https://docs.xarray.dev/en/stable/) are naturally suited. In contrast, Lagrangian data is acquired by observing platforms that move with the flow they are embedded in, for example uncrewed platforms, vehicles, virtual particles, atmospheric phenomena such as tropical cyclones, and even animals that gather data along their natural but complex paths. Because such paths traverse both spatial and temporal dimensions, Lagrangian data often convolve spatial and temporal information that cannot consistently and readily be organized, catalogued, and stored in common data structures and file formats with the help of common libraries and standards. As an example, the concepts of dimensions and coordinates for Lagrangian data are ambiguous and not clearly established. As such, for both data creators and data users, Lagrangian data present challenges that the Clouddrift Python library aims to overcome. 
```
